### PR TITLE
feat(service): added support for labels on serviceSpec

### PR DESCRIPTION
**Pull request by thecodeassassin**
Tailscale services require labels to assign a ProxyClass.

This PR adds support for labels on serviceSpec to accomplish this goal.

### CODE REVIEW

File: api/v1alpha1/dragonfly_types.go [hunk 1]
```diff
@@ -145,6 +145,11 @@ type ServiceSpec struct {
 	// +optional
 	// +kubebuilder:validation:Optional
 	Annotations map[string]string `json:"annotations,omitempty"`
+
+	// (Optional) Dragonfly Service Annotations
```
Comment (Abhra303) [inline api/v1alpha1/dragonfly_types.go]: Update the comment please :)

File: manifests/crd.yaml
Comment (Abhra303) [inline manifests/crd.yaml:1]: can you please drop changes under `manifests` directory to ensure that users don't install these changes before the next release?

I will update these files when I release the next version.

**Review by Abhra303 — COMMENTED** [review]
Thanks for the PR! Mostly looks good. See some comments.

File: api/v1alpha1/dragonfly_types.go (hunk 1 above)
Comment (thecodeassassin) [reply api/v1alpha1/dragonfly_types.go]: Fixed :)

File: manifests/crd.yaml
Comment (thecodeassassin) [reply manifests/crd.yaml:1]: Reverted :)

**Comment by thecodeassassin** [comment]
@Abhra303 Should be good now, changes in manifests reverted

File: manifests/crd.yaml [hunk 2]
```diff
@@ -1604,4 +1604,4 @@ spec:
     served: true
     storage: true
     subresources:
-      status: {}
+      status: {}
```
Comment (Abhra303) [inline manifests/crd.yaml]: I guess you removed the trailing newline in both these files. Please add the trialing newline.

File: manifests/dragonfly-operator.yaml [hunk 3]
```diff
@@ -1978,4 +1978,4 @@ spec:
       securityContext:
         runAsNonRoot: true
       serviceAccountName: dragonfly-operator-controller-manager
-      terminationGracePeriodSeconds: 10
+      terminationGracePeriodSeconds: 10
```
Comment (Abhra303) [inline manifests/dragonfly-operator.yaml]: same, could you please add the trailing newline?

Comment (thecodeassassin) [reply manifests/dragonfly-operator.yaml]: sure, just added them.

Weird, my editor must have removed them?

File: config/crd/bases/dragonflydb.io_dragonflies.yaml [hunk 4]
```diff
@@ -1130,6 +1129,11 @@ spec:
                       type: string
                     description: (Optional) Dragonfly Service Annotations
                     type: object
+                  labels:
+                    additionalProperties:
+                      type: string
+                    description: (Optional) Dragonfly Service Annotations
```
Comment (Abhra303) [inline config/crd/bases/dragonflydb.io_dragonflies.yaml]: Sorry I didn't point it earlier, can you change this too to match with the `Labels` comment ? I will merge once done.

Comment (thecodeassassin) [reply config/crd/bases/dragonflydb.io_dragonflies.yaml]: Sorry, fixed

**Review by Abhra303 — APPROVED** [review]

**Comment by Abhra303** [comment]
Could you please sign the commits so that I can merge?

**Comment by thecodeassassin** [comment]
> Could you please sign the commits so that I can merge?

I am sorry, i don't know what you mean.

**Comment by thecodeassassin** [comment]
I think that should be done ? Can you please check? never signed commits before.

**Comment by Abhra303** [comment]
Looks like you didn't remove the old commits that are not signed, can you please remove those?

Also you should use your full name to sign ( I see only `Stephen` here). You can set your full name by `git config user.name "Your FullName"`. Similarly use your github registered email to set user.email (I see its `stephen@maxroll.gg`). Then sign the commits again. Additionally you can also set PGP key to sign commits (See https://codex.so/gpg-verification-github). Signing with PGP keys verify that its really you and not an imposter signing on behalf of you. Its preferred to have the PGP key.

> never signed commits before.

No issuse, take your time. Its really important to sign your commits (most of the open source repos require signed commits). Once you set up this, its very easy to sign commits (just add a `-S` flag while committing changes). Don't hesitate to ask me if you are stuck anywhere (e.g. PGP setup/ signing, removing the old unsigned commits etc.).

**Comment by thecodeassassin** [comment]
@

> Looks like you didn't remove the old commits that are not signed, can you please remove those?
> 
> Also you should use your full name to sign ( I see only `Stephen` here). You can set your full name by `git config user.name "Your FullName"`. Then sign the commits again. Additionally you can also set PGP key to sign commits (See https://codex.so/gpg-verification-github). Signing with PGP keys verify that its really you and not an imposter signing on behalf of you. Its preferred to have the PGP key.
> 
> > never signed commits before.
> 
> No issuse, take your time. Its really important to sign your commits (most of the open source repos require signed commits). Once you set up this, its very easy to sign commits (just add a `-S` flag while committing changes). Don't hesitate to ask me if you are stuck anywhere (e.g. PGP setup/ signing, removing the old unsigned commits etc.).

Allright, should be good now

**Comment by Abhra303** [comment]
Thanks! It will work!

**Pull request merged**